src/test/java/com/google/error_analyzer/backend/PaginationServletTest.java

```diff
@@ -0,0 +1,179 @@
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+package com.google.error_analyzer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.error_analyzer.backend.LogDao;
+import com.google.error_analyzer.backend.LogDaoHelper;
+import com.google.error_analyzer.servlets.PaginationServlet;
+import java.io.*;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.runners.JUnit4;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;	
+import org.mockito.InjectMocks;	
+import org.mockito.junit.MockitoJUnit;	
+import org.mockito.junit.MockitoRule;	
+import org.mockito.Mock;	
+import org.mockito.Mockito.*;	
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import static org.mockito.BDDMockito.any;
+import static org.mockito.Mockito.when;	
+
+
+@RunWith(MockitoJUnitRunner.class)
+public final class PaginationServletTest {
+
+    private final String fileName = "file";
+    private final String fileType1 = "errors";
+    private final String fileType2 = "logs";
+    private final int page1 = 1;
+    private final int page2 = 2;
+
+    
+    @Mock
+    HttpServletRequest request;
+
+    @Mock
+    HttpServletResponse response;
+
+    @Mock
+    LogDao logDao;
+
+    @Mock
+    LogDaoHelper logDaoHelper;
+
+    @InjectMocks
+    PaginationServlet pagination;
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Before
+    public void setUp() {
+        pagination = new PaginationServlet();
+        ReflectionTestUtils.setField(pagination, "logDao", logDao);
+        ReflectionTestUtils.setField(pagination, "logDaoHelper", 	
+        logDaoHelper);
+        MockitoAnnotations.initMocks(this);
+    }
+
+    // test dopost
+    @Test
+    public void doPost_validLogFileName() throws Exception {	
+        ImmutableList<String> hitContent = 	
+            ImmutableList.of("log2");	
+        when(logDao.fileExists(any(String.class))).thenReturn(true);	
+        when(logDaoHelper.hitFieldContent(any(),any())).	
+            thenReturn(hitContent);
+        when(request.getParameter("start")).thenReturn("1");
```

Comment:
After creating "start" etc. as constants in LogFields, use those constants here.